spec/facility_factory_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'spec_helper'
+
+describe FacilityFactory do
+  before(:each) do
+    @facility_factory = FacilityFactory.new
+  end
+
+  describe '#initialize' do
+    it 'is an instance of FacilityFactory' do
+      expect(@facility_factory).to be_an_instance_of(FacilityFactory)
+    end
+  end
+
+  describe '#create_facilites' do
+    it 'creates facilities using stored API data' do
+      co_dmv_info = DmvDataService.new.co_dmv_office_locations
+      created_facilities = @facility_factory.create_facilities(co_dmv_info)
+
+      expect(co_dmv_info.count).to eq(5)
+      expect(created_facilities.count).to eq(5)
+
+      expect(created_facilities).to be_an(Array)
+
+      created_facilities.each do |facility|
+        expect(facility).to be_an_instance_of(Facility)
+      end
+
+      expect(created_facilities[0].name).to eq("DMV Tremont Branch")
+      expect(created_facilities[0].address).to eq("2855 Tremont Place Suite 118 Denver CO 80205") #has address line 2 in data
+      expect(created_facilities[0].phone).to eq("(720) 865-4600")
+
+      expect(created_facilities[2].name).to eq("DMV Northwest Branch")
+      expect(created_facilities[2].address).to eq("3698 W. 44th Avenue Denver CO 80211") #does not have address line 2 in data
+      expect(created_facilities[2].phone).to eq("(720) 865-4600")
+    end
+    
+    it 'can create facilities using data sets with different formats' do
+      ny_dmv_info = DmvDataService.new.ny_dmv_office_locations
+      mo_dmv_info = DmvDataService.new.mo_dmv_office_locations
+
+      expect(ny_dmv_info.count).to eq(173)
+      expect(mo_dmv_info.count).to eq(178)
```

Comment:
Did not think about the fact that API data can change. Should have used:
expect(mo_dmv_info.count == mo_facilities.count).to be true